src/main/java/com/sap/sgs/phosphor/fosstars/model/Confidence.java

```diff
@@ -66,28 +69,60 @@ static double adjust(double value) {
   }
 
   /**
-   * Calculates a confidence level based on a number of unknown feature values.
+   * Calculates a confidence level based on a number of values.
+   * The methods takes into account the following:
+   * <ul>
+   *   <li>number of unknown values</li>
+   *   <li>score values with their confidences and weights</li>
+   *   <li>values with confidences</li>
+   * </ul>
    *
    * @param values The features values.
    * @return The confidence level.
    */
-  static double make(Value... values) {
+  static double make(List<Value> values) {
     Objects.requireNonNull(values, "Hey! Values can't be null!");
-    if (values.length == 0) {
+
+    if (values.size() == 0) {
       throw new IllegalArgumentException("Hey! Values can't be empty!");
     }
-    int unknown = 0;
+
+    double weightSum = 0.0;
+    double weightedConfidenceSum = 0.0;
+
     for (Value value : values) {
       if (value.isUnknown()) {
-        unknown++;
+
+        // if value is unknown, then confidence is min and weight is 1.0
+        weightSum += 1.0;
+      } else if (value instanceof ScoreValue) {
+        ScoreValue scoreValue = (ScoreValue) value;
+        weightSum += scoreValue.weight();
+        weightedConfidenceSum += scoreValue.weight() * scoreValue.confidence();
+      } else if (value instanceof Confidence) {
+        weightedConfidenceSum += ((Confidence) value).confidence();
+        weightSum += 1.0;
+      } else {
+
```

Comment:
Please remove this extra line.

Reply:
Actually I intentionally added a line here. See also above. For some reason, an extra line before a comment in the beginning of a block looks nicer to me. I know, it's quite subjective. I can remove it if it bothers.